packages/back-end/src/models/ProjectModel.ts

```diff
@@ -1,139 +1,79 @@
   },
-  name: String,
-  description: String,
-  dateCreated: Date,
-  dateUpdated: Date,
-  settings: {},
+  projectScoping: "none",
+  globallyUniqueIds: true,
 });
 
-type ProjectDocument = mongoose.Document & ProjectInterface;
-
-const ProjectModel = mongoose.model<ProjectInterface>("Project", projectSchema);
-
-function toInterface(doc: ProjectDocument): ProjectInterface {
-  const ret = doc.toJSON<ProjectDocument>();
-  ret.settings = ret.settings || {};
-  return omit(ret, ["__v", "_id"]);
-}
-
 interface CreateProjectProps {
   name: string;
   description?: string;
   id?: string;
 }
 
-export async function createProject(
-  organization: string,
-  data: CreateProjectProps
-) {
-  const doc = await ProjectModel.create({
-    organization: organization,
-    id: data.id || uniqid("prj_"),
-    name: data.name || "",
-    description: data.description,
-    dateCreated: new Date(),
-    dateUpdated: new Date(),
-  });
-  return toInterface(doc);
-}
-export async function findAllProjectsByOrganization(
-  context: ReqContext | ApiReqContext
-) {
-  const { org } = context;
-  const docs = await ProjectModel.find({
-    organization: org.id,
-  });
+export class ProjectModel extends BaseClass {
+  protected canRead(doc: ProjectInterface) {
+    return this.context.permissions.canReadSingleProjectResource(doc.id);
+  }
 
-  const projects = docs.map(toInterface);
-  return projects.filter((p) =>
-    context.permissions.canReadSingleProjectResource(p.id)
-  );
-}
-export async function findProjectById(
-  context: ReqContext | ApiReqContext,
-  projectId: string
-) {
-  const { org } = context;
-  const doc = await ProjectModel.findOne({
-    id: projectId,
-    organization: org.id,
-  });
-  if (!doc) return null;
+  protected canCreate() {
+    return this.context.permissions.canCreateProjects();
+  }
 
-  const project = toInterface(doc);
+  protected canUpdate(doc: ProjectInterface) {
+    return this.context.permissions.canUpdateProject(doc.id);
+  }
 
-  return context.permissions.canReadSingleProjectResource(project.id)
-    ? project
-    : null;
-}
-export async function deleteProjectById(id: string, organization: string) {
-  await ProjectModel.deleteOne({
-    id,
-    organization,
-  });
-}
-export async function updateProject(
-  id: string,
-  organization: string,
-  update: Partial<ProjectInterface>
-) {
-  await ProjectModel.updateOne(
-    {
-      id,
-      organization,
-    },
-    {
-      $set: update,
-    }
-  );
-}
+  protected canDelete(doc: ProjectInterface) {
+    return this.context.permissions.canDeleteProject(doc.id);
+  }
 
-export async function updateProjectSettings(
-  id: string,
-  organization: string,
-  settings: Partial<ProjectSettings>
-) {
-  const update = {
-    $set: {
-      dateUpdated: new Date(),
-      settings,
-    },
-  };
-  await ProjectModel.updateOne(
-    {
-      id,
-      organization,
-    },
-    update
-  );
-}
+  protected migrate(doc: z.infer<typeof projectValidator>) {
```

Comment:
The type here isn't quite accurate and is making Typescript hide potential errors.  Some old documents don't have `settings` at all, and some that do don't have `statsEngine` defined.